specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -3956,6 +4152,329 @@
+				},
+				"type": "string",
+				"description": "The name of the skillset."
+			},
+			"description": {
+				"type": "string",
+				"description": "The description of the skillset."
+			},
+			"skills": {
+				"type": "array",
+				"items": {
+					"$ref": "#/definitions/Skill"
+				},
+				"description": "A list of skills in the skillset."
+			},
+			"@odata.etag": {
+				"x-ms-client-name": "ETag",
+				"type": "string",
+				"description": "The ETag of the skillset."
+			}
+		},
+		"required": ["name",
+		"description",
+		"skills"],
+		"description": "A list of Cognitive Search skills."
+	},
+	"Skill": {
+		"discriminator": "@odata.type",
+		"properties": {
+			"@odata.type": {
+				"type": "string"
+			},
+			"description": {
+				"type": "string",
+				"description": "The description of the skill."
+			},
+			"context": {
+				"type": "string",
+				"description": "The context of the skill."
+			},
+			"inputs": {
+				"type": "array",
+				"items": {
+					"$ref": "#/definitions/InputFieldMappingEntry"
+				},
+				"description": "The inputs of the skill."
+			},
+			"outputs": {
+				"type": "array",
+				"items": {
+					"$ref": "#/definitions/OutputFieldMappingEntry"
+				},
+				"description": "The outputs of the skill."
+			}
+		},
+		"required": ["@odata.type",
+		"description",
+		"context",
+		"inputs",
+		"outputs"],
+		"description": "Abstract base class for skills."
+	},
+	"InputFieldMappingEntry": {
+		"properties": {
+			"name": {
+				"type": "string",
+				"description": "The name of the input."
+			},
+			"source": {
+				"type": "string",
+				"description": "The source of the input."
+			}
+		},
+		"required": ["name",
+		"source"],
+		"description": "Input field mapping for a skill."
+	},
+	"OutputFieldMappingEntry": {
+		"properties": {
+			"name": {
+				"type": "string",
+				"description": "The name of the output."
+			},
+			"targetName": {
+				"type": "string",
+				"description": "The target name of the output."
+			}
+		},
+		"required": ["name",
+		"targetName"],
+		"description": "output field mapping for a skill."
+	},
+	"KeyPhraseExtractionSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Text.KeyPhraseExtractionSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"properties": {
+			"defaultLanguageCode": {
+				"type": "string",
+				"default": "en",
+				"description": "A value indicating which language code. Default is en."
+			},
+			"maxKeyPhraseCount": {
+				"type": "integer",
+				"format": "int32",
+				"x-nullable": true
+			}
+		},
+		"description": "Text analytics key phrase extraction."
+	},
+	"OcrSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Vision.OcrSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"properties": {
+			"textExtractionAlgorithm": {
+				"type": "string",
+				"default": "printed",
+				"description": "A value indicating which algorithm to use. Default is printed."
```

Comment:
While this correctly indicates that the property is formatted as a string on the wire, it doesn't really help to create a robust client-side developer experience. Whenever there is a property like this that can take on one of a limited set of values, we should define an enum for it. For properties where the value set is very strict and unlikely to change, we can model it as a normal enum (see `TokenCharacterKind` as an example). For properties where the set of values is more open-ended and may change between API versions, we should model them as extensible enums. Currently this requires custom code. See `AnalyzerName` as an example.

Reply:
Added enums.